src/packages/api/api_manager.py

```diff
@@ -0,0 +1,468 @@
+                              attr['data']['markedForDeletion']['reporter'],
+                              attr['data']['markedForDeletion']['reason'])
+        quotes = [Quotation(quote['message'],
+                            quote['author'],
+                            APIManager.parse_api_datetime(quote['createdAt']),
+                            APIManager.parse_api_datetime(quote['updatedAt']),
+                            quote['lastAuthor'])
+                  for quote in attr['quotes']]
+        rats = []
+        if included:
+            for include in included:
+                if include['type'] != 'rats':
+                    continue
+                rats.append(Rat(UUID(include['id']),
+                                include['attributes']['name'],
+                                Platforms[include['attributes']['platform'].upper()]))
+        first_limpet = None
+        if data['relationships']['firstLimpet']['data']:
+            first_limpet = data['relationships']['firstLimpet']['data']['id']
+        rescue = Rescue(
+            uuid=UUID(data['id']),
+            client=attr['client'],
+            code_red=attr['codeRed'],
+            lang_id=attr['data']['langID'],
+            irc_nickname=attr['data']['IRCNick'],
+            board_index=attr['data']['boardIndex'],
+            mark_for_deletion=mfd,
+            platform=Platforms[attr['platform'].upper()],
+            quotes=quotes,
+            status=Status[attr['status'].upper()],
+            system=attr['system'],
+            title=attr['title'],
+            unidentified_rats=attr['unidentifiedRats'],
+            rats=rats,
+            first_limpet=first_limpet,
+            created_at=APIManager.parse_api_datetime(attr['createdAt']),
+            updated_at=APIManager.parse_api_datetime(attr['updatedAt'])
+        )
+        # rescue.outcome = attr['outcome']  # @TODO Needs a setter or something...
+        return rescue
+
+    def __init__(self, config: typing.Optional[typing.Dict]):
+        if config:
+            self.rehash_handler(config)
+
+    async def get_rescue(self, uuid: str) -> Rescue:
+        """
+        Use the API to get a rescue by UUID.
+
+        Args:
+            uuid (str): The Rescue's UUID.
+
+        Returns:
+            A Rescue object describing the Rescue found, or None if none was found.
+        """
+        if not uuid:
+            raise ValueError("uuid must be specified!")
+
+        try:
+            data = await self._call(f"rescues/{uuid}", {'include': 'rats'})
+            if data['data']:
+                return self._deserialize_rescue(data['data'], data['included'])
+        except aiohttp.ClientError:
+            return None
+
+    async def create_rescue(self, rescue: Rescue):
+        """
+        Create the case in the API.
+
+        Args:
+            rescue (Rescue): The Rescue object to create in the API.
+        """
+        data = APIManager._serialize_rescue(rescue)
+        del data['id']
```

Comment:
[question] why are we deleting this field?

Reply:
As of right now, APIv3 will throw a hissy-fit if you try to create a Rescue with a pre-made UUID.
That very well might change when we move away from the mockup, but that's how it works now.

Reply:
Yeah, thats gunna produce a problem given UUIDs are immutable in our data model under the understanding mecha is the one generating the UUIDs 😕